src/test/kotlin/org/jmailen/gradle/kotlinter/functional/ModifiedSourceSetsTest.kt

```diff
@@ -0,0 +1,129 @@
+package org.jmailen.gradle.kotlinter.functional
+
+import java.io.File
+import org.gradle.testkit.runner.TaskOutcome
+import org.intellij.lang.annotations.Language
+import org.jmailen.gradle.kotlinter.functional.utils.androidManifest
+import org.jmailen.gradle.kotlinter.functional.utils.kotlinClass
+import org.jmailen.gradle.kotlinter.functional.utils.resolve
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+internal class ModifiedSourceSetsTest : WithGradleTest.Android() {
+
+    private lateinit var androidModuleRoot: File
+    private lateinit var kotlinModuleRoot: File
+
+    @Before
+    fun setUp() {
+        testProjectDir.root.apply {
+            resolve("settings.gradle") { writeText(settingsFile) }
+            resolve("build.gradle") {
+                @Language("groovy")
+                val buildScript = """
+                subprojects {
+                    repositories {
+                        google()
+                        jcenter()
+                    }
+                }
+                
+            """.trimIndent()
+                writeText(buildScript)
+            }
+            androidModuleRoot = resolve("androidproject") {
+                resolve("build.gradle") {
+                    @Language("groovy")
+                    val androidBuildScript = """
+                        plugins {
+                            id 'com.android.library'
+                            id 'kotlin-android'
+                            id 'org.jmailen.kotlinter'
+                        }
+                        
+                        android {
+                            compileSdkVersion 29
+                            defaultConfig {
+                                minSdkVersion 23
+                            }
+                            sourceSets {
```

Comment:
Is it possible to test an _added_ sourceSet and see if we pick that up too? I don't know how common it is for people to actually do it, but would be nice for completeness.

Reply:
Huh, I don't know if it is possible to add custom sourceset that would be recognized by AGP 🤔 
If you add that as you would for `java` plugin a _"The SourceSet 'xxxxxx' is not recognized by the Android Gradle Plugin. Perhaps you misspelled something?"_ message pops up. 

Given that, I've added new SourceSet for Kotlin project and for flavor specific SourceSet for Android project.